STACpopulator/implementations/CMIP6_UofT/add_CMIP6.py

```diff
@@ -69,15 +71,15 @@ def create_stac_item(self, item_name: str, item_data: MutableMapping[str, Any])
             dc_ext.apply(dimensions=dc_helper.dimensions, variables=dc_helper.variables)
         except Exception:
             LOGGER.error("Failed to add Datacube extension to item %s", item_name)
-            raise
+            return (-1, "Failed to add Datacube extension")
 
         try:
             thredds_helper = THREDDSHelper(item_data["access_urls"])
             thredds_ext = THREDDSExtension.ext(item)
             thredds_ext.apply(thredds_helper.services, thredds_helper.links)
         except Exception:
             LOGGER.error("Failed to add THREDDS references to item %s", item_name)
-            raise
+            return (-1, "Failed to add THREDDS references")
```

Comment:
This and other above should instead make use of pythonic `raise <StacPopulatorException>("message")`. 
The parent operation then decides what to do with the exception.